aiosmtpd/tests/test_server.py

```diff
@@ -56,3 +73,143 @@ def test_server_attribute(self):
         finally:
             controller.stop()
             self.assertIsNone(controller.server)
+
+
+# Silence the "Exception ignored ... RuntimeError: Event loop is closed" message.
+# This goes hand-in-hand with TestFactory.setUpClass() below.
+# Source: https://github.com/aio-libs/aiohttp/issues/4324#issuecomment-733884349
+def silencer(func):
+    @wraps(func)
+    def wrapper(self, *args, **kwargs):
+        try:
+            return func(self, *args, **kwargs)
+        except RuntimeError as e:
+            if str(e) != "Event loop is closed":
+                raise
+        except AttributeError as e:
+            # Added this to suppress the perplexing ignored exception below.
+            # Perplexing because I really don't know where that comes from. SMTP object
+            # could NOT have existed during that particular test case!
+            if str(e) != "'SMTP' object has no attribute '_closed'":
+                raise
+
+    return wrapper
+
+
+class TestFactory(unittest.TestCase):
+    Proactor = None
+    old_pa_del = None
+    SRProto = None
+    old_srp_del = None
+
+    @classmethod
+    def setUpClass(cls) -> None:
+        # See silencer() above
+        try:
+            # noinspection PyUnresolvedReferences
+            cls.Proactor = asyncio.proactor_events._ProactorBasePipeTransport
+            cls.old_pa_del = cls.Proactor.__del__
+            cls.Proactor.__del__ = silencer(cls.old_pa_del)
+        except AttributeError:
+            # proactor_events only available on Windows. So we'll just skip if it's
+            # not found (indicating non-Windows platform)
+            pass
+        try:
+            cls.SRProto = asyncio.streams.StreamReaderProtocol
+            # noinspection PyUnresolvedReferences
+            cls.old_srp_del = cls.SRProto.__del__
+            cls.SRProto.__del__ = silencer(cls.old_srp_del)
+        except AttributeError:
+            # Sometimes the __del__ method is ... missing?? What???
+            pass
+
+    @classmethod
+    def tearDownClass(cls) -> None:
+        # gc.collect() hinted in https://stackoverflow.com/a/25067818/149900
+        # Probably to remove leftover "Exception ignored"?
+        gc.collect()
```

Comment:
Without `gc.collect()` somehow the "Exception ignored" bit survived capture in `silencer()`. Not sure why.